app/client.jsx

```diff
@@ -14,11 +15,25 @@ const store = configureStore(initialState, browserHistory);
 const history = syncHistoryWithStore(browserHistory, store);
 const routes = createRoutes(store);
 
+/**
+ * Callback function handling frontend route changes.
+ */
+function onUpdate() {
+  // Prevent duplicate fetches when first loaded.
+  if (window.__INITIAL_STATE__ !== null) {
+    window.__INITIAL_STATE__ = null;
```

Comment:
Thanks for that, I get what you mean now.

We only need `window.__INITITAL_STATE__` for configuring the store on server-rendered pages.

The check is only for preventing duplicate fetches when `server-render` -> client side kicks in and doesn't need to refetch. 

Every subsequent client `onUpdate` call should then trigger a `fetchComponentDataBeforeRender` because `__INITIAL__STATE__` is outdated. 👍